File: core/src/main/java/com/emarsys/core/util/log/entry/OfflineQueueSize.java
package com.emarsys.core.util.log.entry;

import java.util.HashMap;
import java.util.Map;

public class OfflineQueueSize implements LogEntry {
    private final int queueSize;

    public OfflineQueueSize(int queueSize) {
        this.queueSize = queueSize;
    }

    @Override
    public String getTopic() {
        return "log_offline_queue_size";
    }

    @Override
    public Map<String, Object> getData() {
        Map<String, Object> data = new HashMap<>();
        data.put("queue_size", queueSize);
        return data;
    }
}

File: core/src/main/java/com/emarsys/core/util/log/entry/OfflineQueueSize.kt
package com.emarsys.core.util.log.entry

class OfflineQueueSize(queueSize: Int) : LogEntry {
    override val data: Map<String, Any>
    override val topic: String
        get() = "log_offline_queue_size"

    init {
        data = mapOf(
                "queue_size" to queueSize
        )
    }
}